Lib/test/test_str.py

```diff
@@ -1908,6 +1909,11 @@ def test_utf8_decode_invalid_sequences(self):
             self.assertRaises(UnicodeDecodeError,
                               (b'\xF4'+cb+b'\xBF\xBF').decode, 'utf-8')
 
+    def test_issue127903(self):
+        # Issue #127903: segmentation fault in debug mode in ``_copy_characters``
+        # when there is nothing to copy.
+        self.assertEqual(datetime.datetime(2013, 11, 10, 14, 20, 59).strftime('%z'), '')
```

Comment:
```suggestion
        # Issue #127903: ``_copy_characters`` crashes on DEBUG builds when
        # there is nothing to copy.
        d = datetime.datetime(2013, 11, 10, 14, 20, 59)
        self.assertEqual(d.strftime('%z'), '')
```

Reply:
(The suggestion is to show that the the crash is not in the construction of the datetime object)